src/librustc_codegen_llvm/mir/block.rs

```diff
@@ -507,14 +507,40 @@ impl<'a, 'tcx> FunctionCx<'a, 'tcx> {
                         // promotes any complex rvalues to constants.
                         if i == 2 && intrinsic.unwrap().starts_with("simd_shuffle") {
                             match *arg {
+                                // The shuffle array argument is usually not an explicit constant,
+                                // but specified directly in the code. This means it gets promoted
+                                // and we can then extract the value by evaluating the promoted.
+                                mir::Operand::Copy(mir::Place::Promoted(box(index, ty))) |
+                                mir::Operand::Move(mir::Place::Promoted(box(index, ty))) => {
+                                    let param_env = ty::ParamEnv::reveal_all();
```

Comment:
This should probably be something like `self.param_env` (minor future hazard if we start deduplicating instances).

Reply:
The PR already landed.Does anyone take care of fixing this?^^

Reply:
This is rustc_codegen_llvm... we don't have a param env, everything is monomorphized